### PR TITLE
add preview docs section

### PR DESCRIPTION
**Pull request by epassaro**
## Description
Add a documentation section on how to build docs in your fork to preview changes (and check you are not breaking the docs!).

## Motivation and Context
Sometimes it's useful to share a preview of changes in documentation. But the most important thing is this will help to make the pipeline break less often (if users are encouraged to preview changes in documentation before merging).

## How Has This Been Tested?
See: https://github.com/epassaro/tardis/actions/
and https://epassaro.github.io/tardis/development/documentation_preview.html

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have assigned/requested two reviewers for this pull request.

### CODE REVIEW

File: docs/development/pr_docs.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+2. Edit ``.github/workflows/documentation-build.yml`` and add a new trigger below the *push* trigger:
+
+.. code-block: none
+    pull_request:
+      branches:
+        - master
+
+3. Push changes and make a new pull request to **your fork's** *master* branch.
+
+4. Go to your fork's *Settings* tab make sure GitHub Pages are building from the *gh-pages* branch.
```
Comment (andrewfullard) [inline docs/development/pr_docs.rst:33]: This should be first, I think

File: docs/development/pr_docs.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+5. If everything is ok, the documentation preview should be available at ``<your-username>.github.io/tardis``.
+
+6. Remember you will need to rebase ``new-feature-docs`` to ``new-feature`` every time you push changes to ``new-feature``.
```
Comment (andrewfullard) [inline docs/development/pr_docs.rst:37]: Bold this

Comment (epassaro) [reply docs/development/pr_docs.rst]: Done @andrewfullard

File: docs/development/pr_docs.rst (hunk 1 above)
Comment (epassaro) [reply docs/development/pr_docs.rst]: Done

**Review by andrewfullard — APPROVED** [review]

**Pull request merged**